app/uk/gov/hmrc/economiccrimelevyreturns/controllers/AmlRegulatedActivityController.scala

```diff
@@ -53,19 +58,42 @@ class AmlRegulatedActivityController @Inject() (
   }
 
   def onSubmit(mode: Mode): Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
+    implicit val hc: HeaderCarrier = CorrelationIdHelper.getOrCreateCorrelationId(request)
     form
       .bindFromRequest()
       .fold(
         formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, request.startAmendUrl))),
-        amlRegulatedActivity =>
-          eclReturnsConnector
-            .upsertReturn(
-              dataCleanup
-                .cleanup(request.eclReturn.copy(carriedOutAmlRegulatedActivityForFullFy = Some(amlRegulatedActivity)))
+        carriedOutAmlRegulatedActivityForFullFy => {
+          val eclReturn =
+            dataCleanup.cleanup(
+              request.eclReturn.copy(
+                carriedOutAmlRegulatedActivityForFullFy = Some(carriedOutAmlRegulatedActivityForFullFy)
+              )
             )
-            .flatMap { updatedReturn =>
-              pageNavigator.nextPage(mode, updatedReturn).map(Redirect)
-            }
+
+          (for {
+            unit <- eclReturnsService.upsertReturn(eclReturn).asResponseError
+          } yield unit)
+            .foldF(
+              err => Future.successful(Status(err.code.statusCode)(Json.toJson(err))),
```

Comment:
How does this error message manifest? We should be showing the default error page, but this looks like it'll show JSON like what's happening in Account server.